src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -243,26 +244,39 @@ public SyntaxTriviaList Insert(int index, SyntaxTrivia trivia)
         /// <param name="trivia">The trivia to insert.</param>
         public SyntaxTriviaList InsertRange(int index, IEnumerable<SyntaxTrivia> trivia)
         {
-            if (index < 0 || index > this.Count)
+            var thisCount = this.Count;
+            if (index < 0 || index > thisCount)
             {
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
 
-            var items = trivia.ToList();
-            if (items.Count == 0)
+            if (trivia == null)
+            {
+                throw new ArgumentNullException(nameof(trivia));
+            }
+
+            // Just return ourselves if we're not being asked to add anything.
+            var triviaCollection = trivia as ICollection;
```

Comment:
I've moved to ```ICollection<T>```.  Some of our own types implement that, but not ICollection (like SpecializedCollections.EmptyList).  ImmutableArray also does not implement ICollection.  So it seems like most of the types we care about are going to be using the generic version.